src/holdem/parse.rs

```diff
@@ -522,6 +523,33 @@ impl RangeParser {
 
         Ok(filtered)
     }
+
+    /// Parse a string and return all the starting hands
+    ///
+    /// # Examples
+    ///
+    /// Same as `parse_one` but this will parse a comma separated list of hands.
+    ///
+    /// ```
+    /// use rs_poker::holdem::RangeParser;
+    ///
+    /// let hand = RangeParser::parse("KK+,A2s+").unwrap();
+    /// assert_eq!(60, hand.len());
+    ///
+    /// // Filters out duplicates.
+    /// assert_eq!(RangeParser::parse_one("AK-87s,A2s+").unwrap().len(), 72)
+    /// ```
+    pub fn parse(r_str: &str) -> Result<Vec<Hand>, RSPokerError> {
+        let mut set: HashSet<Hand> = HashSet::new();
+
+        for hand_str in r_str.split(',') {
+            let hand = RangeParser::parse_one(hand_str.trim())?;
+
+            set.extend(hand)
+        }
```

Comment:
I think this is better written as split -> map(parse_one) -> collect into HashSet -> into_iter -> collect as vec()

See https://doc.rust-lang.org/rust-by-example/error/iter_result.html#fail-the-entire-operation-with-collect For error handling.

Reply:
I changed the implementation to try and follow your suggestions. I am still relatively new to rust, so let me know if this needs improvement.